parsec/interfaces/dtd/insert_function.c

```diff
@@ -2303,8 +2303,8 @@ static parsec_hook_return_t parsec_dtd_gpu_task_submit(parsec_execution_stream_t
     parsec_dtd_task_class_t *dtd_tc = (parsec_dtd_task_class_t*)this_task->task_class;
 
     dev_index = parsec_get_best_device(this_task, &load);
-    assert(dev_index >= 0);
-    if (dev_index < 2) {
+    assert(dev_index >= 0 && dev_index < parsec_mca_device_enabled());
+    if (parsec_mca_device_get(dev_index)->type <= PARSEC_DEV_RECURSIVE) {
```

Comment:
Comparing with `PARSEC_DEV_RECURSIVE` is still not great, IMHO. For one, it's not intuitive as it relies on a specific ordering of device identifier. Second, it's error prone (e.g., if the order changes) and verbose (which is probably the reason for the short-cut `< 2`).

Could we have masks to check for device classes? Or a function `parsec_mca_device_is_accel(dev_index)`?